core/services/run_manager.go

```diff
@@ -264,6 +264,10 @@ func (jm *runManager) Create(
 		return nil, errors.Wrap(err, "CreateJobRun failed")
 	}
 
+	if runRequest.RequestID != nil {
```

Comment:
Would it make sense to put this in the `New run triggered` stanza above and make that Infow?
You could add the requestId logging after like so: 
```go
details := []interface{"job", ..., }
if runRequest.RequestID != nil {
  details = append(details, []interface{"externalID", ...
}
logger.Infow("...", details)
```

Reply:
I want a line where the two IDs can be tied together, and the run ID isn't available in that line. We do a similar thing in the `run.ForLogger` method, so I added it in there.